spec/section_spec.rb

```diff
@@ -4,37 +4,56 @@
 
 describe SitePrism::Page do
   describe '.section' do
+    class Page < SitePrism::Page; end
+    class Section < SitePrism::Section; end
+
+    class PageWithAnonymousSection < SitePrism::Page
+      section :anonymous_section, '.section' do |s|
+        s.element :title, 'h1'
+      end
+    end
+
+    class PageWithSection < SitePrism::Page
+      section :section, Section, '.section'
+    end
+
+    context 'second argument is a Class' do
+      subject { PageWithSection.new }
+
+      it 'should create a section using the Class' do
+        expect(subject).to respond_to(:section)
+      end
+    end
+
     context 'second argument is not a Class and a block given' do
-      context 'block given' do
-        it 'should create an anonymous section with the block' do
-          class PageWithSection < SitePrism::Page
-            section :anonymous_section, '.section' do |s|
-              s.element :title, 'h1'
-            end
-          end
-
-          page = PageWithSection.new
-          expect(page).to respond_to(:anonymous_section)
-        end
+      subject { PageWithAnonymousSection.new }
+
+      it 'should create an anonymous section using the block' do
+        expect(subject).to respond_to(:anonymous_section)
       end
     end
 
-    context 'second argument is not a class and no block given' do
-      subject(:section) { Page.section(:incorrect_section, '.section') }
+    context 'second argument is not a Class and no block given' do
+      subject { Page.section(:incorrect_section, '.section') }
+      let(:error_message) do
+        'You should provide section class either as a block, or as the second argument.'
+      end
 
       it 'should raise an ArgumentError' do
-        class Page < SitePrism::Page; end
-
-        expect { section }
+        expect { subject }
           .to raise_error(ArgumentError)
-          .with_message('You should provide section class either as a block, or as the second argument.')
+          .with_message(error_message)
       end
     end
   end
 end
 
 describe SitePrism::Section do
-  let(:a_page) { class Page < SitePrism::Page; end }
+  class Page < SitePrism::Page; end
```

Comment:
I mean the parent class of a Section being a SitePrism::Page is definitely the assumed usage even if it may not be strictly necessary for this group of tests.